### PR TITLE
Add hotkeys for squash merging

### PR DESCRIPTION
**Pull request by vanniktech**
This is a proposal for adding a shortcut on the following buttons:

screen shot 2018-03-10 at 22 11 22

screen shot 2018-03-10 at 22 12 02

When pressing `a s` both buttons will be pressed and the branch will be squash merged. `a` for action since `m` and other shortcuts are already taken from Github.

This is a prototype just to get some information whether you folks want to include this or not. We can extend this for the other 2 types.

### CODE REVIEW

**Comment by vanniktech** [comment]
I'll fix the style issues once I got some feedback on the idea.

File: source/features/add-squash-branch-hotkey.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	const squashButton = select(`.btn-group-squash > button`);
+
+	if (squashButton) {
+		squashButton.setAttribute('data-hotkey', 'a s');
+	}
+
+  const confirmButton = select(`.btn-group-squash > button[name="do"]`);
+
+  if (confirmButton) {
+    confirmButton.setAttribute('data-hotkey', 'a s');
```
Comment (sindresorhus) [inline source/features/add-squash-branch-hotkey.js]: Use `.dataset` instead.

**Comment by sindresorhus** [comment]
I would use this.

**Comment by sindresorhus** [comment]
> This is a prototype just to get some information whether you folks want to include this or not. We can extend this for the other 2 types.

I would wait until someone requests it. I think 95% of cases this one is what people want.

**Comment by sindresorhus** [comment]
Also need to do this: https://github.com/sindresorhus/refined-github/pull/1167#issuecomment-372911882

File: source/features/add-squash-branch-hotkey.js (hunk 1 above)
Comment (vanniktech) [reply source/features/add-squash-branch-hotkey.js]: Done

**Comment by vanniktech** [comment]
Thanks for the feedback. I adjusted everything.

**Comment by hkdobrev** [comment]
Why not just open the form with the shortcut, focusing the commit title and you can execute with cmd+enter? This way you can do everything from the keyboard, but still have the confirmation and possibility to keep the same flow of events and edit the commit.

**Comment by hkdobrev** [comment]
Also, I'd suggest the shortcut to use the default merge strategy set for the repo, no matter if it's squash or merge.

**Comment by vanniktech** [comment]
On what type of interaction should we agree? There are quite a lot of different opinions about this feature.

**Comment by sindresorhus** [comment]
> Why not just open the form with the shortcut, focusing the commit title and you can execute with cmd+enter? This way you can do everything from the keyboard, but still have the confirmation and possibility to keep the same flow of events and edit the commit.

Yes, let's go with this.

**Comment by bfred-it** [comment]
Can this be picked up again?

**Comment by vanniktech** [comment]
I was planning to but I'm super busy and likely won't be able to start picking this up in the next two weeks.

**Comment by vanniktech** [comment]
I'll close this since it's not that likely I'll continue this - anytime soon. I also ran into some local problems when using node and trying to get the plugin to work.

**Pull request closed without merging**